### ISSUE TITLE
Player is not destroyed if Quaver is timed out while queuing tracks

### ISSUE THREAD

**Issue by Xayanide** [ZPTXDev/Quaver#385]
**Describe the bug**
What isn't working as intended, and what does it affect?
Affects functionality

**Severity**
- [x] Critical
- [ ] High
- [ ] Medium
- [ ] Low

**Affected versions**
What versions are affected by this bug? (e.g. >=3.0.1, 2.5.1-2.6.3, >=1.2.0)
`> 3.3.0`

**Steps to reproduce**
Steps to reproduce the behavior. (e.g. click on a button, enter a value, etc. and see error)
1. Queue any track with Quaver with play or search command in a voice or stage channel.
2. After queueing the tracks or sending the command, quickly time out Quaver on the spot.

**Expected behavior**
What is expected to happen?
Cancel the interaction, destroy the player.

**Actual behavior**
What actually happens? Attach or add errors or screenshots here as well.
There is no error.
Quaver didn't destroy the player when it got timed out. The player still exists while Quaver is nowhere to be found on any voice or stage channels.

Whenever the user uses commands that involves players. Quaver will say
`You have to be in my voice channel`

Basically Quaver thinks it is in a voice or stage channel when it is not.
